### ISSUE TITLE
Position shifting each time the mesh goes back and forth between Zbrush and Blender

### ISSUE THREAD

**Issue by Tsukasa-YAMATO** [JoseConseco/GoB#119]
![2020-06-22_17h36_46](https://user-images.githubusercontent.com/65609416/85268804-f8868d00-b4b1-11ea-8002-d5bf6c1c6a8b.jpg)

Environment
-------
Zbrush 2020.1.4
GoB Ver.3.3.9
Blender 2.83

---------

Reproduction method
-------
1. Move back and forth between Zbrush and Blender (mesh may be created on either side first)
2. Move or rotate the mesh "in Object mode" in Blender (Rotation 30 degrees is the most obvious example)
3. Reciprocating Zbrush and Blender many times
4. The moved or rotated mesh gradually shifts

Cause
-------
When transferring a mesh from Zbrush to Blender with GoB, if the mesh has the object mode transpose (Rotation and/or position offset) in Blender, misalignment occurs after import finished. 

This is a problem on the Blender side, and there is nothing wrong with Zbrush side. The position of the world origin of Zbrush does not change, but it seems that the origin when importing GoB in Blender is not the world origin but the local origin of the object (orange point). Therefore, in case that the origin of the object is not set to the world origin of Blender, when GoB imports from Zbrush to Blender, it will be set to  different position from Zbrush.
Applying "the all transforms" resets the object's local origin to Blender's world origin, so the Zbrush's world origin and Blender's local origin are the same position, so no misalignment occurs.

![2020-06-22_18h08_37](https://user-images.githubusercontent.com/65609416/85269879-7c8d4480-b4b3-11ea-8f35-53cd1e6a910d.jpg)

So the first time you transfer a mesh from Zbrush to Blender, no misalignment will occur. And also the problem does not occur even if the mesh is processed in the "edit mode" on Blender or the mesh is deformed by "rig deformation". Because the both are NOT transformations by object mode( It means that all offset and rotation are kept to 0 and not be changed.).

The principle of solution
-------
When transferring a mesh from Zbrush to Blender with GoB, it is only necessary to import the target object in Blender with "all transforms" being applied. That is, the rotation and position offsets are all set to 0, and after that the mesh should be imported, it's no problem.

Solution
-------
Adding a function to automatically apply "all transforms" in the operation of GoB importing a mesh from Zbrush (I hope this function can be turned ON/OFF by the user as an option. Default:ON)
In other words, after the GoB import is complete, all imported objects will be applyed "all transforms". (Rotation and position offset are all 0.)

Now it can be avoided if the user manually applies all the transforms and then imports GoB, but it is preferable to implement the Solution because it prevents unnecessary bugs and misalignment due to user mistakes.

Please consider the implementation.

**Comment by kromar** [comment]
Thanks for the in-depth report, this is outstanding!
I think this should be fixed in this branch: https://github.com/kromar/blender_GoB/tree/bugfix/crash-and-position-change
would be great if you could try it and let me know if it fixes the transformation issues you encountered.

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
Can you change the version number? For example, Ver.3.3.9.1.

I'm starting to lose control over whether the version number I'm testing now is a stable 3.3.9 or something else on the branch. If the version number shown in the Blender's preferences is 3.3.9.1, etc., it will be easier to communicate.
Right now, I'm starting to lose track of which version is "Ver.3.3.9".

**Comment by kromar** [comment]
sure i will update the branch to 3.3.10

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
Thank you for your quick response.
I have just confirmed it now.
Please see the image.

![2020-06-23_03h16_04](https://user-images.githubusercontent.com/65609416/85322057-ad479b00-b500-11ea-9195-1df01cec3f41.jpg)

It is clear that good communication is possible.
Thank you!

In the future, if there are a minor change and you publish it, you should change the numbers. Otherwise, misunderstanding may occur unexpectedly.

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
With GoB Ver.3.3.10, it works without any misalignment. I was surprised  and glad at the smarter implementation than my suggestion. :)
Great!
I will use Ver.3.3.10 for a while.

**Comment by kromar** [comment]
great to hear that this solved your rotation issue as well!
